lib/jxl/encode.cc

```diff
@@ -191,6 +207,21 @@ JxlEncoderStatus JxlEncoderSetBasicInfo(JxlEncoder* enc,
       return JXL_ENC_ERROR;
       break;
   }
+  if (info->have_animation) {
+    enc->metadata.m.have_animation = true;
```

Comment:
This would better be outside of the if and be:

enc->metadata.m.have_animation = info->have_animation

because if enc->metadata.m.have_animation is true, but user now uses SetBasicInfo with have_animation = false, enc->metadata.m.have_animation should be set to false again too.